incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -271,33 +263,69 @@ func (r *ObjectReconciler) ignore(ctx context.Context, log logr.Logger, inst *un
 		log.V(1).Info("Containing namespace hasn't been synced yet")
 		return true
 	}
-	// If the namespace has critical condition, do nothing.
-	if ns.HasCritCondition() {
-		log.V(1).Info("Containing namespace has critical condition(s)")
-		return true
-	}
 
 	return false
 }
 
-// syncObject handles a source object and a propagated object differently. If a source object changes,
-// all descendant copies will be enqueued. If a propagated object is obsolete, it will be deleted.
-// Otherwise, it will be overwritten by the source if they are different.
-func (r *ObjectReconciler) syncObject(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) (action, *unstructured.Unstructured) {
+// syncMissingObject handles the case where the object we're reconciling doesn't exist. If it can
+// determine a final action to take, it returns that action, otherwise it returns actionUnknown
+// which indicates that we need to call the regular syncObject method. Note that this method may
+// modify `inst`.
+func (r *ObjectReconciler) syncMissingObject(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) syncAction {
+	// If the object exists, skip
+	if inst.GetCreationTimestamp() != (v1.Time{}) {
```

Comment:
Replace it with `!inst.GetCreationTimestampe().isZero()` to be consistent across?

Reply:
This is one of those annoying cases where we can't call IsZero() because it expects a pointer receiver, and `inst.GetCreationTimestamp()` doesn't return one. See #292 .

Reply:
Thanks for filing that issue earlier for easy referencing now!